lib/rdstation/api_response.rb

```diff
@@ -5,5 +5,11 @@ def self.build(response)
 
       RDStation::ErrorHandler.new(response).raise_error
     end
+
+    def self.build_with_headers(response)
+      return response if response.code.between?(200, 299)
+
+      RDStation::ErrorHandler.new(response).raise_error
+    end
   end
 end
```

Comment:
<!-- Suggestions are specific requests to improve the subject under review. It is assumed that we all want to do what's best, so these comments are never dismissed as “mere suggestions”, but are taken seriously.  -->
**🎯 suggestion:** ‏What do you think about changing the `RdStation::ApiResponse.build` to accept an option argument to indicate whether is necessary to return with headers?

Something like this:

```ruby
# frozen_string_literal: true

module RDStation
  module ApiResponse
    def self.build(response, options={})
      successful = response.code.between?(200, 299)
      RDStation::ErrorHandler.new(response).raise_error unless successful

      if options[:headers]
        response
      else
        JSON.parse(response.body)
      end
    end
  end
end

# usage
RDStation::ApiResponse.build(response, headers: true)

```

Reply:
But, let me ask something, what is returned with the `response` object? The original (current) `build` method returns a JSON and the new one returns an object "response".

I guess we're changing the API contract, don't we?

What do you think @angeliski ?